site/src/components/pages/user/UserPageComponent.tsx

```diff
@@ -0,0 +1,350 @@
+
+  const { user: currentUser } = useUser();
+
+  const isCurrentUserPage = useMemo(() => {
+    if (currentUser !== "loading" && currentUser) {
+      return currentUser.id === user.id;
+    }
+    return false;
+  }, [user, currentUser]);
+
+  const handleSchemaTitleChange = (value: string) => {
+    let formattedText = value.trim();
+    // replace all empty spaces
+    formattedText = formattedText.replace(/\s/g, "");
+
+    // capitalize text
+    if (formattedText.length > 1) {
+      formattedText = formattedText[0].toUpperCase() + formattedText.slice(1);
+    }
+
+    setNewSchemaTitle(formattedText);
+  };
+
+  const handleCreateSchema = useCallback(
+    async (evt: FormEvent) => {
+      evt.preventDefault();
+      if (newSchemaTitle === "") {
+        setError("Please enter a valid value");
+        return;
+      }
+
+      setLoading(true);
+      const { data, error: apiError } = await apiClient.createEntityType({
+        schema: {
+          title: newSchemaTitle,
+        },
+      });
+      setLoading(false);
+      if (apiError) {
+        if (apiError.response?.data.errors) {
+          setError(apiError.response.data.errors[0].msg);
+        } else {
+          // @todo properly handle this
+          setError("An error occured");
+        }
+      } else {
+        const schemaTitle = data?.entityType.schema.title;
+        void router.push(`/@${user.shortname}/types/${schemaTitle}`);
+      }
+    },
+    [user, newSchemaTitle, router],
+  );
+
+  return (
+    <>
+      <Head>
+        <title>{user.shortname}</title>
+      </Head>
+      <Divider
+        sx={{
+          borderColor: ({ palette }) => palette.gray[20],
+        }}
+      />
+      <Box
+        sx={{
+          background: ({ palette }) => palette.gray[10],
+          flex: 1,
+          display: "flex",
+          flexDirection: "column",
+          minHeight: "80vh",
+          pb: 10,
+          mt: { xs: 4, md: 10 },
+        }}
+      >
+        <Container
+          sx={{
+            display: "flex",
+            flexDirection: { xs: "column", md: "row" },
+            flex: 1,
+            justifyContent: "flex-start",
+            maxWidth: "1200px !important",
+          }}
+        >
+          {/* SIDEBAR */}
+          <Box
+            sx={{
+              width: { xs: "100%", md: SIDEBAR_WIDTH },
+              mr: { xs: 0, md: 8 },
+              background: {
+                xs: theme.palette.common.white,
+                md: "transparent",
+              },
+              pb: 8,
+            }}
+          >
+            <Sidebar isMobile={isMobile} user={user} />
+          </Box>
+          {/* CONTENT */}
+          <Box
+            sx={{
+              flex: 1,
+            }}
+          >
+            {/* TAB HEADER */}
+            <TabHeader
+              activeTab={activeTab}
+              setActiveTab={(nextTab) => {
+                setActiveTab(nextTab);
+                return router.push(
```

Comment:
Routing between tabs happens here

Reply:
This doesn't need a `return`

Reply:
@CiaranMn this is for satisfying `no-floating-promises`.

Reply:
Understood - this is not really in the spirit of `no-floating-promises` because this is moving the promise to be unhandled elsewhere, rather than actually handling it. But not important in this case.